pkg/provider/azure_vmssflex_cache_test.go

```diff
@@ -0,0 +1,507 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-07-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	cloudprovider "k8s.io/cloud-provider"
+
+	"sigs.k8s.io/cloud-provider-azure/pkg/azureclients/vmclient/mockvmclient"
+	"sigs.k8s.io/cloud-provider-azure/pkg/azureclients/vmssclient/mockvmssclient"
+	azcache "sigs.k8s.io/cloud-provider-azure/pkg/cache"
+	"sigs.k8s.io/cloud-provider-azure/pkg/retry"
+)
+
+var (
+	testVMWithoutInstanceView1 = compute.VirtualMachine{
+		Name: to.StringPtr("testvm1"),
+		VirtualMachineProperties: &compute.VirtualMachineProperties{
+			ProvisioningState: nil,
+			VirtualMachineScaleSet: &compute.SubResource{
+				ID: to.StringPtr("subscriptions/sub/resourceGroups/rg/providers/Microsoft.Compute/virtualMachineScaleSets/vmssflex1"),
+			},
+		},
+	}
+
+	testVMWithoutInstanceView2 = compute.VirtualMachine{
+		Name: to.StringPtr("testvm2"),
+		VirtualMachineProperties: &compute.VirtualMachineProperties{
+			ProvisioningState: nil,
+			VirtualMachineScaleSet: &compute.SubResource{
+				ID: to.StringPtr("subscriptions/sub/resourceGroups/rg/providers/Microsoft.Compute/virtualMachineScaleSets/vmssflex2"),
+			},
+		},
+	}
+	testVMListWithoutInstanceView       = []compute.VirtualMachine{testVMWithoutInstanceView1, testVMWithoutInstanceView2}
+	testVMListResultWithoutInstanceView = compute.VirtualMachineListResult{
+		Value: &testVMListWithoutInstanceView,
+	}
+
+	testVMWithOnlyInstanceView1 = compute.VirtualMachine{
+		Name: to.StringPtr("testvm1"),
+		VirtualMachineProperties: &compute.VirtualMachineProperties{
+			InstanceView: &compute.VirtualMachineInstanceView{
+				Statuses: &[]compute.InstanceViewStatus{
+					{
+						Code: to.StringPtr("PowerState/running"),
+					},
+				},
+			},
+		},
+	}
+
+	testVMWithOnlyInstanceView2 = compute.VirtualMachine{
+		Name: to.StringPtr("testvm2"),
+		VirtualMachineProperties: &compute.VirtualMachineProperties{
+			InstanceView: &compute.VirtualMachineInstanceView{
+				Statuses: &[]compute.InstanceViewStatus{
+					{
+						Code: to.StringPtr("PowerState/running"),
+					},
+				},
+			},
+		},
+	}
+	testVMListWithOnlyInstanceView       = []compute.VirtualMachine{testVMWithOnlyInstanceView1, testVMWithOnlyInstanceView2}
+	testVMListResultWithOnlyInstanceView = compute.VirtualMachineListResult{
```

Comment:
testVMListResultWithOnlyInstanceView unused?

Reply:
You are right, it is not used. I just removed it. Thanks for catching this.